frontend/console/src/features/modules/decls/TypeEl.tsx

```diff
@@ -0,0 +1,66 @@
+import type { Optional, Ref, Array as SchArray, Map as SchMap, Type } from '../../../protos/xyz/block/ftl/v1/schema/schema_pb'
+import { DeclLink } from './DeclLink'
+
+// TypeParams ironically is built to work with the `Type` message, not the
+// `TypeParameter` message, which just has a simple string param name without
+// any higher level type information.
+const TypeParams = ({ types }: { types?: (Type | undefined)[] }) => {
+  if (!types) {
+    return []
+  }
+  const definedTypes = types.filter((t) => t !== undefined)
+  if (definedTypes.length === 0) {
+    return []
+  }
+  return (
+    <span>
+      <span>{'<'}</span>
+      {definedTypes.map((t, i) => [<TypeEl key='t' t={t} />, i === types.length - 1 ? '' : ', '])}
```

Comment:
should this be using `definedTypes.length` here? instead of `types.length`?

Also, I think this can be refactored to take advantage of optionals.

```ts
const TypeParams = ({ types }: { types?: (Type | undefined)[] }) => {
  const definedTypes = types?.filter((t) => t !== undefined) || []
  if (definedTypes.length === 0) return
    
  return (
    <span>
      <span>{'<'}</span>
      {definedTypes.map((t, i) => [<TypeEl key='t' t={t} />, i === definedTypes.length - 1 ? '' : ', '])}
      <span>{'>'}</span>
    </span>
  )
}
```